src/monmove.cpp

```diff
@@ -418,16 +418,42 @@ void monster::move()
     int mondex = !plans.empty() ? g->mon_at( plans[0], is_hallucination() ) : -1;
```

Comment:
This should probably call `critter_at` so it can recognize the player character and any NPCs as well.

This would also allow to get rid of the explicit usage of `g->u` at line 424.